runtime/transformations/components/filters/correct_time_intervals_with_fixed_values.py

```diff
@@ -0,0 +1,237 @@
+        },
+        {
+            "start": "2020-01-03T08:20:02.000Z",
+            "end": "2020-01-03T08:20:04.000Z",
+            "start_inclusive": false,
+            "end_inclusive": false,
+            "correction_value": 42.0
+        }
+    ]
+}
+```
+Output of the above call is:
+```
+{
+    "corrected":{
+        "__hd_wrapped_data_object__":"SERIES",
+        "__metadata__":{},
+        "__data__":{
+            "2020-01-01T01:15:27.000Z":37.0,
+            "2020-01-03T08:20:03.000Z":42.0,
+            "2020-01-03T08:20:04.000Z":25.9
+        }
+    }
+}
+```
+"""
+
+import pandas as pd
+from pydantic import BaseModel, ValidationError, root_validator
+
+from hetdesrun.runtime.exceptions import ComponentInputValidationException
+
+
+class TimeInterval(BaseModel):
+    start: str
+    end: str
+    correction_value: float
+    start_inclusive: bool = True
+    end_inclusive: bool = True
+
+    @root_validator()
+    def verify_value_ranges(cls, values: dict) -> dict:
+        start = values["start"]
+        start_inclusive = values["start_inclusive"]
+        end = values["end"]
+        end_inclusive = values["end_inclusive"]
+        if end < start:
+            raise ValueError(
+                "To be valid, a time interval must be non-empty, i.e. the start timestamp "
+                "may not be bigger than the end timestamp."
+            )
+        if (start_inclusive is False or end_inclusive is False) and not (start < end):
+            raise ValueError(
+                "To be valid, a time interval must be non-empty, i.e the start timestamp must be "
+                "smaller than the end timestamp."
+            )
+        return values
+
+
+# ***** DO NOT EDIT LINES BELOW *****
+# These lines may be overwritten if component details or inputs/outputs change.
+COMPONENT_INFO = {
+    "inputs": {
+        "series": {"data_type": "SERIES"},
+        "list_of_time_intervals": {"data_type": "ANY"},
+    },
+    "outputs": {
+        "corrected": {"data_type": "SERIES"},
+    },
+    "name": "Correct time intervals with fixed values",
+    "category": "Filters",
+    "description": (
+        "Correct all data points in provided time intervals from a time series with a "
+        "correction value for each time interval."
+    ),
+    "version_tag": "1.0.0",
+    "id": "504a01aa-12a2-472b-995c-0676af66b07c",
+    "revision_group_id": "92d2ded2-c6ca-42a1-a15b-a8e9cf6ca635",
+    "state": "RELEASED",
+    "released_timestamp": "2023-11-09T08:04:31.787859+00:00",
+    "disabled_timestamp": None,
+}
+
+
+def main(*, series: pd.Series, list_of_time_intervals):
+    # entrypoint function for this component
+    # ***** DO NOT EDIT LINES ABOVE *****
+    # write your function code here.
+
+    if isinstance(series, pd.Series) and isinstance(series.index, pd.DatetimeIndex):
+        timestamps = series.index
+    else:
+        raise ComponentInputValidationException(
+            "Could not find timestamps in provided object",
```

Comment:
Clearer message, something like: The index of the provided series does not have the datatype DatetimeIndex